packages/devtools_app/lib/src/utils.dart

```diff
@@ -1069,3 +1069,34 @@ String prettyTimestamp(
   );
   return DateFormat.Hms().format(timestampDT); // HH:mm:ss
 }
+
+/// A [ChangeNotifier] that holds a list of data.
+///
+/// When [value] is replaced with something that is not equal to the old
+/// value as evaluated by the equality operator ==, this class notifies its
+/// listeners.
+///
+/// This class also exposes methods `addAndNotify` and `addAllAndNotify` that
+/// will notify listeners upon the list [value] being modified.
+class ListValueNotifier<T> extends ValueNotifier<List<T>> {
+  /// Creates a [ListValueNotifier] that wraps this value.
+  ListValueNotifier(List<T> value) : super(value);
+
+  void add(T item) {
+    value.add(item);
+    notifyListeners();
+  }
+
+  void addAll(List<T> itemList) {
+    value.addAll(itemList);
+    notifyListeners();
+  }
+
```

Comment:
clear() would be useful (especially when user clear all all collected memory/snapshot data) as would removeAt() might be too other removes I'd add if needed later.

Reply:
added clear